### PR TITLE
Going with the Flow - Switching from Iterable to Flow

### PR DESCRIPTION
**Pull request by idugalic**

### CODE REVIEW

File: application/src/main/kotlin/com/fraktalio/fmodel/application/MaterializedView.kt [hunk 1]
```diff
@@ -44,11 +44,16 @@ data class MaterializedView<S, E>(
...
-            val newState = view.evolve(oldState, event)
-            newState.save().bind()
+        try {
+            // Arrow provides a Monad instance for Either. Except for the types signatures, our program remains unchanged when we compute over Either. All values on the left side assume to be Right biased and, whenever a Left value is found, the computation short-circuits, producing a result that is compatible with the function type signature.
+            either {
+                val oldState = event.fetchStateEither().bind() ?: view.initialState
+                val newState = view.evolve(oldState, event)
+                newState.saveEither().bind()
+            }
+        } catch (ex: Exception) {
```
[the remark this answers is not in the document]
Comment (raulraja) [reply application/src/main/kotlin/com/fraktalio/fmodel/application/MaterializedView.kt:54]: Here https://arrow-kt.io/docs/apidocs/arrow-core/arrow.core/kotlin.-throwable/index.html#extensions-for-kotlinthrowable may help 👋

[the remark this answers is not in the document]
Comment (idugalic) [reply application/src/main/kotlin/com/fraktalio/fmodel/application/MaterializedView.kt:54]: Thanks! I do not want to catch them all...

File: application/src/main/kotlin/com/fraktalio/fmodel/application/EventSourcingAggregate.kt [hunk 2]
```diff
@@ -52,36 +52,44 @@ data class EventSourcingAggregate<C, S, E>(
...
+            if (saga != null)
+                resultingEvents.flatMapConcat { saga.react(it) }.collect {
+                    val newEvents = flowOf(currentEvents, resultingEvents).flattenConcat().calculateNewEvents(it)
+                    resultingEvents = flowOf(resultingEvents, newEvents).flattenConcat()
                 }
-            newEvents
-        }
 
+            return resultingEvents
+        } catch (e: Throwable) {
```
[the remark this answers is not in the document]
Comment (idugalic) [reply application/src/main/kotlin/com/fraktalio/fmodel/application/EventSourcingAggregate.kt:90]: ignore

File: application/src/main/kotlin/com/fraktalio/fmodel/application/SagaManager.kt [hunk 3]
```diff
@@ -35,12 +39,30 @@ data class SagaManager<AR, A>(
...
+     * @param actionResults Action Results represent the outcome of some action you want to handle in some way
+     * @return [Flow] of [Either] [Error] or [A]/Action
+     */
+    fun handle(actionResults: Flow<AR>): Flow<Either<Error, A>> =
+        actionResults.flatMapConcat { handle(it) }
+
+    private fun AR.calculateNewActions(): Flow<A> =
+        try {
+            saga.react(this)
+        } catch (e: Throwable) {
```
[the remark this answers is not in the document]
Comment (idugalic) [reply application/src/main/kotlin/com/fraktalio/fmodel/application/SagaManager.kt:61]: ignore

File: application/src/main/kotlin/com/fraktalio/fmodel/application/EventSourcingAggregate.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (idugalic) [reply application/src/main/kotlin/com/fraktalio/fmodel/application/EventSourcingAggregate.kt:90]: @sonatype-lift ignore

File: application/src/main/kotlin/com/fraktalio/fmodel/application/SagaManager.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (idugalic) [reply application/src/main/kotlin/com/fraktalio/fmodel/application/SagaManager.kt:61]: @sonatype-lift ignore

File: application/src/main/kotlin/com/fraktalio/fmodel/application/EventSourcingAggregate.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (idugalic) [reply application/src/main/kotlin/com/fraktalio/fmodel/application/EventSourcingAggregate.kt:90]: @sonatype-lift ignore

**Comment by idugalic** [comment]
@sonatype-lift help

**Pull request merged**